packages/parsers/openapi/shared/composable/availability.node.ts

```
@@ -0,0 +1,35 @@
+import { ApiNode, ApiNodeContext, ComposableApiNode } from "../interfaces/api.node.interface";
+
+export class AvailabilityNode<T, InputNode extends ApiNode<unknown, T> & { availability: string }>
+    implements
+        ComposableApiNode<
+            T,
+            InputNode,
+            T & {
+                availability: string;
+            }
+        >
+{
+    name = "availability";
+    preProcessedInput: InputNode["preProcessedInput"];
+    qualifiedId: string;
+
+    constructor(
+        readonly context: ApiNodeContext,
+        readonly inputNode: InputNode,
+        readonly accessPath: ApiNode<unknown, unknown>[],
+    ) {
+        this.qualifiedId = `${this.accessPath.map((node) => node.qualifiedId).join(".")}.${this.name}`;
+        this.preProcessedInput = this.inputNode.preProcessedInput;
+    }
+
+    outputFdrShape(): T & {
+        availability: string;
+    } {
+        const baseShape = this.inputNode.outputFdrShape();
+        return {
+            ...baseShape,
+            availability: this.inputNode.availability,
+        };
+    }
```

Thoughts on how you are planning to handle errors/excdeptions as you do the conversion? 

It could be helpful if every output FdrShape was wrapped in a discriminated union that had a success/error return value. This might be nice so that even if someone has a misconfigured part of their OpenAPI spec, you dont have to fail hard. 

```
ApiOutput = SuccessApiOutput | ErrorApiOutput; 

interface SuccessApiOutput {
  type: "success"; 
  result: FdrShape
}

interface ErrorOutput {
  type: "error";
  message: string;
}
```

---

Another option could be creating an `ErrorCollector` where you can collect validation errors as part of the converter.

---

I think the error collector makes sense -- agreed on not failing hard -- we will have to stop the processing for any of the applicable descendants, but we should definitely get as wide of a parsing view as possible

---

Even if you have an error collector the return value needs to be `undefined` or OutputShape